### PR TITLE
fix (issue 966): ensure domain socket for spire agent is of mode 775

### PR DESCRIPTION
**Pull request by malnick**
**Pull Request check list**

- [x] Commit conforms to CONTRIBUTING.md?
- [ ] Proper tests/regressions included?
- [ ] Documentation updated?

**Affected functionality**
Please see https://github.com/spiffe/spire/issues/966

**Description of change**
Enable permissive umask when creating the domain socket for the spire agent. 

**Which issue this PR fixes**
https://github.com/spiffe/spire/issues/966

### CODE REVIEW

**Comment by azdagron** [comment]
Thanks for the contribution, @malnick!

I'm concerned about switching the umask, even temporarily, after things have started up in the agent since the umask is process wide and other goroutines could accidentally create a file/directory without the umask in place.

I think I'd rather see the MkdirAll done from run/run.go before the umask is manipulated (maybe with the logging that @APTy suggested in the issue).

**Comment by azdagron** [comment]
Also, we require DCO on all commits (you can follow the details link on the failed DCO check for instructions on how to amend).

**Comment by malnick** [comment]
@azdagron I pushed a fresh commit, let me know if this is more aligned with what you were thinking.

File: cmd/spire-agent/cli/run/run.go [hunk 1]
```diff
@@ -96,6 +96,15 @@ func (*RunCLI) Run(args []string) int {
 		return 1
 	}
 
+	// Create uds dir and parents if not exists
+	dir := filepath.Dir(c.BindAddress.String())
+	c.Log.Infof("creating spire agent UDS path %s", dir)
```
Comment (azdagron) [inline cmd/spire-agent/cli/run/run.go]: Logging should start with capital letter (i.e. `"Creating UDS directory"`)

Prefer fields for parameterized data (keeps the logs clean and easier to read) (i.e. `c.Log.WithField("dir", dir).Info("Creating UDS directory")`)

File: cmd/spire-agent/cli/run/run.go [hunk 2]
```diff
@@ -96,6 +96,15 @@ func (*RunCLI) Run(args []string) int {
 		return 1
 	}
 
+	// Create uds dir and parents if not exists
+	dir := filepath.Dir(c.BindAddress.String())
```
Comment (azdagron) [inline cmd/spire-agent/cli/run/run.go]: What would you think stat'ing the directory first before doing the log+mkdirall and only proceeding if the directory does not exist? The logging might be less confusing (I'd be wondering why it was logging that it was creating the directory when I knew it already existed).

Comment (malnick) [reply cmd/spire-agent/cli/run/run.go]: Ha, that was going to be my next move after I confirmed the code location. Sounds good :+1:

**Comment by malnick** [comment]
@azdagron New commit pushed to address the PR comments. Let me know if this needs anything else and thanks for the quick responses!

**Pull request merged**